src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalState = { ...state };
+  const stateHistory = [];
+
+  for (const order of actions) {
+    const ADD_PROPERTIES = order.type === 'addProperties';
+    const REMOVE_PROPERTIES = order.type === 'removeProperties';
+    const CLEAR_ALL = order.type === 'clear';
```

Comment:
please move those outside of function 
```suggestion
    const ADD_PROPERTIES = 'addProperties';
    const REMOVE_PROPERTIES = 'removeProperties';
    const CLEAR_ALL = 'clear';
```